xarray/core/dataset.py

```diff
@@ -5146,5 +5149,106 @@ def filter_by_attrs(self, **kwargs):
                 selection.append(var_name)
         return self[selection]
 
+    def unify_chunks(self):
+        """ Unifies chunksize along all chunked dimensions of this Dataset.
+
+        Returns
+        -------
+
+        Dataset with consistent chunk sizes for all dask-array variables
+
+        See Also
+        --------
+
+        dask.array.core.unify_chunks
+        """
+        import dask.array
+
+        try:
+            if self.chunks:
+                return self
```

Comment:
This will subtly break user code if the user relies on this method to always be a shallow copy.
Replace with ``self.copy()``